File: packages/shared/src/types/enums.ts
/**
 * Shared enum/type alias definitions for Beach League.
 * These must match the backend's Python enums in models.py.
 */

/** Player's personal gender identity. null = not specified / prefer not to say. */
export type PlayerGender = 'male' | 'female';

/**
 * League / tournament gender division.
 * Aligned with KoB tournament conventions (migration 038).
 */
export type LeagueGender = 'mens' | 'womens' | 'coed';

/** Canonical skill level values. */
export type SkillLevel = 'juniors' | 'beginner' | 'intermediate' | 'advanced' | 'AA' | 'Open';

/** Session lifecycle status (maps to backend SessionStatus enum). */
export type SessionStatus = 'ACTIVE' | 'SUBMITTED' | 'EDITED';

/**
 * Court moderation status.
 * NOTE: The backend DB column uses 'approved' (not 'active') for live courts.
 */
export type CourtStatus = 'pending' | 'approved' | 'rejected';

/** League member role values. */
export type LeagueMemberRole = 'admin' | 'member';

/** KOB tournament scheduling format (maps to backend TournamentFormat enum). */
export type KobTournamentFormat = 'FULL_ROUND_ROBIN' | 'POOLS_PLAYOFFS' | 'PARTIAL_ROUND_ROBIN';

/** KOB tournament lifecycle status (maps to backend TournamentStatus enum). */
export type KobTournamentStatus = 'SETUP' | 'ACTIVE' | 'COMPLETED' | 'CANCELLED';

/** Friend request lifecycle status. */
export type FriendRequestStatus = 'pending' | 'accepted' | 'rejected';

/**
 * Notification type values — must match backend NotificationType enum in models.py.
 */
export type NotificationType =
  | 'league_message'
  | 'league_invite'
  | 'league_join_request'
  | 'league_join_rejected'
  | 'season_start'
  | 'season_activated'
  | 'placeholder_claimed'
  | 'friend_request'
  | 'friend_accepted'
  | 'session_submitted'
  | 'session_auto_submitted'
  | 'session_auto_deleted'
  | 'member_joined'
  | 'member_removed'
  | 'direct_message'
  | 'season_award';

File: apps/backend/database/models/enums.py
"""Enums models."""

import enum


class SessionStatus(str, enum.Enum):
    """Session status enum."""

    ACTIVE = "ACTIVE"
    SUBMITTED = "SUBMITTED"
    EDITED = "EDITED"


class OpenSignupsMode(str, enum.Enum):
    """Weekly schedule signup opening mode."""

    AUTO_AFTER_LAST_SESSION = "auto_after_last_session"
    SPECIFIC_DAY_TIME = "specific_day_time"
    ALWAYS_OPEN = "always_open"


class SignupEventType(str, enum.Enum):
    """Signup event type."""

    SIGNUP = "signup"
    DROPOUT = "dropout"


class ScoringSystem(str, enum.Enum):
    """Season scoring system enum."""

    POINTS_SYSTEM = "points_system"
    SEASON_RATING = "season_rating"


class FriendRequestStatus(str, enum.Enum):
    """Friend request status enum."""

    PENDING = "pending"
    ACCEPTED = "accepted"
    DECLINED = "declined"
    CANCELLED = "cancelled"
    SUPERSEDED = "superseded"


class NotificationType(str, enum.Enum):
    """Notification type enum."""

    LEAGUE_MESSAGE = "league_message"
    LEAGUE_INVITE = "league_invite"
    LEAGUE_JOIN_REQUEST = "league_join_request"
    LEAGUE_JOIN_REJECTED = "league_join_rejected"
    SEASON_START = "season_start"
    SEASON_ACTIVATED = "season_activated"
    PLACEHOLDER_CLAIMED = "placeholder_claimed"
    FRIEND_REQUEST = "friend_request"
    FRIEND_ACCEPTED = "friend_accepted"
    SESSION_SUBMITTED = "session_submitted"
    SESSION_AUTO_SUBMITTED = "session_auto_submitted"
    SESSION_AUTO_DELETED = "session_auto_deleted"
    MEMBER_JOINED = "member_joined"
    MEMBER_REMOVED = "member_removed"
    DIRECT_MESSAGE = "direct_message"
    SEASON_AWARD = "season_award"
    MODERATION_UPDATE = "moderation_update"


class InviteStatus(str, enum.Enum):
    """Player invite status enum."""

    PENDING = "pending"
    CLAIMED = "claimed"
